File: src/main/java/com/leijendary/spring/boot/template/data/PageMeta.java
package com.leijendary.spring.boot.template.data;

import org.springframework.data.domain.Page;

import lombok.Data;

@Data
public class PageMeta {

    private int numberOfElements;
    private int totalPages;
    private long totalElements;
    private int size;
    private int page;

    public PageMeta(final Page<?> page) {
        this.numberOfElements = page.getNumberOfElements();
        this.totalPages = page.getTotalPages();
        this.totalElements = page.getTotalElements();
        this.size = page.getSize();
        this.page = page.getNumber() + 1;
    }
}

File: src/main/kotlin/com/leijendary/spring/boot/template/core/data/PageMeta.kt
package com.leijendary.spring.boot.template.core.data

import org.springframework.data.domain.Page

class PageMeta(page: Page<*>) {
    val numberOfElements: Int = page.numberOfElements
    val totalPages: Int = page.totalPages
    val totalElements: Long = page.totalElements
    val size: Int = page.size
    val number: Int = page.number
}